torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -47,31 +52,40 @@ TypePtr MergeInferredType(TypePtr existing_type, TypePtr inferred_type) {
     if (!old_tensor_type->device()) {
       // device not avaible means this is an invalid tensor type (most likely an
       // empty one) return inferred type directly.
-      return new_tensor_type;
+      return std::make_pair(new_tensor_type, true);
     }
     auto type = old_tensor_type;
     if (new_tensor_type->dim()) {
       type = type->withSymbolicShapes(new_tensor_type->symbolic_sizes());
+      use_inferred_type = true;
     }
     if (new_tensor_type->scalarType().has_value()) {
       type = type->withScalarType(new_tensor_type->scalarType());
+      use_inferred_type = true;
     }
-    return type;
+    return std::make_pair(type, use_inferred_type);
   }
 
   if (old_tensor_type) {
-    return existing_type;
+    return std::make_pair(existing_type, false);
   }
 
   auto old_list_type = existing_type->cast<ListType>();
   if (new_tensor_type && old_list_type) {
     if (new_tensor_type->sizes().isComplete()) {
-      return inferred_type;
+      return std::make_pair(inferred_type, true);
     }
-    return existing_type;
+    return std::make_pair(existing_type, false);
   }
 
-  return inferred_type;
+  return std::make_pair(inferred_type, true);
+}
+
+void MergeInferredTypeAndSetMap(Value* v, TypePtr inferred_type) {
+  auto input_reliable = IsShapeReliable(v->node());
```

Comment:
`input_reliable` unused. Is it intentionally not updated in the map?

Reply:
The reliable info is updated at the end of shape inference, the logic is like this:
```python
  if (n->kind() != prim::ListConstruct) {
    for (auto input : n->inputs()) {
      if (input->node()->kind() == prim::ListConstruct) {
        UpdateReliable(input, IsShapeReliable(input->node()));
      }
    }
  }
  auto input_reliable = IsShapeReliable(n);
  for (auto output : n->outputs()) {
    UpdateReliable(output, input_reliable);
  }
```
We need first look at whether input is prim::ListConstruct, and then update reliable for the node.
`MergeInferredTypeAndSetMap` may be called more than once, we don't need update reliable every time (seems redundant and we need consider ListConstruct case)